pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -154,6 +152,12 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 	stillRunning := time.NewTicker(25 * time.Second)
 	ps.StillRunning(agentName, warningTime, errorTime)
 
+	if err := utils.WaitForVault(ps, log, agentName, warningTime, errorTime); err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Instead of waiting I think we need to have domainmgr react to VaultStatus changes and when the vault is unlocked, it will start the user containerd. Thus the upgradeconverter will just to the moving of the files.

Otherwise domainmgr will not progress to report AssignableAdapters etc until the vault has been unlocked.
Also, it will be unable to start domains whose volumes live in /persist/clear/  before the vault is unlocked. (I don't think EVE-OS currently does that due to how volumemgr works, but we shouldn't add more obstacles to making it work.)


In addition, one thing we didn't talk about on the call is how containerd will

Reply:
Good point, will modify.
It seems your message ended in the half of sentence for some reason.

Reply:
It needs to be started at the beginning of pillar. I do not know what the right location is though.

Reply:
The main reason to have it (the second containerd) is to run after vault unlocked to use decrypted directories.